create-and-save-map/src/main/java/com/esri/arcgismaps/sample/createandsavemap/components/MapViewModel.kt

```diff
@@ -0,0 +1,178 @@
+/* Copyright 2024 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.createandsavemap.components
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.viewModelScope
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.AndroidViewModel
+
+import com.arcgismaps.httpcore.authentication.OAuthUserConfiguration
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.Basemap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.ArcGISMapImageLayer
+import com.arcgismaps.mapping.layers.Layer
+import com.arcgismaps.portal.Portal
+import com.arcgismaps.portal.PortalFolder
+import com.arcgismaps.toolkit.authentication.AuthenticatorState
+import com.esri.arcgismaps.sample.createandsavemap.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+
+class MapViewModel(application: Application) : AndroidViewModel(application) {
+
+    // view model to handle popup dialogs
+    val messageDialogVM = MessageDialogViewModel()
+
+    // set up authenticator state to handle authentication challenges
+    val authenticatorState = AuthenticatorState().apply {
+        oAuthUserConfiguration = OAuthUserConfiguration(
+            portalUrl = application.getString(R.string.portal_url),
+            clientId = application.getString(R.string.client_id),
+            redirectUrl = application.getString(R.string.redirect_url)
+        )
+    }
+
+    // require use of user credential to load portal
+    private val portal = Portal("https://www.arcgis.com", Portal.Connection.Authenticated)
+
+    // update displayed map once user is authenticated
+    var arcGISMap by mutableStateOf(ArcGISMap())
+
+    // folders on portal associated with the authenticated user
+    private val _portalFolders = MutableStateFlow<List<PortalFolder>>(listOf())
+    val portalFolders = _portalFolders.asStateFlow()
+
+
+    // load a couple of feature layers
```

Comment:
Comment says "load" but it looks like you are just creating layers and they get loaded much further down.